backend/typescript/services/implementations/adminService.ts

```diff
@@ -108,6 +114,52 @@ class AdminService implements IAdminService {
     }
     return form;
   }
+
+  async addQuestionToTemplate(
+    formQuestion: CreateFormQuestionDTO,
+  ): Promise<FormQuestionDTO> {
+    const session = await mongoose.startSession();
+    session.startTransaction();
+
+    try {
+      // Get the current formTemplate and its list of questions
+      const formTemplate = await MgFormTemplate.findOne();
+      if (!formTemplate)
+        throw new Error("Failed to retrieve the form template");
+
+      const newFormQuestion = await MgFormQuestion.create({
+        type: formQuestion.type,
+        question: formQuestion.question,
+        required: formQuestion.required,
+        description: formQuestion.description,
+        options: formQuestion.options,
+        category: formQuestion.category,
+      });
+
+      formTemplate.formQuestions.push(newFormQuestion._id);
+      await formTemplate.save();
+      await session.commitTransaction();
+
+      return {
+        type: newFormQuestion.type,
+        question: newFormQuestion.question,
+        required: newFormQuestion.required,
+        description: newFormQuestion.description,
+        options: newFormQuestion.options,
+        id: newFormQuestion.id,
+        category: newFormQuestion.category,
+      };
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to add form question to form template. Reason = ${getErrorMessage(
+          error,
+        )}`,
+      );
+      await session.abortTransaction();
+      await session.endSession();
```

Comment:
should this be moved to a `finally {}`? Even if no error, we should end session

Reply:
good callout